frontend/pages/hosts/ManageHostsPage/ManageHostsPage.tsx

```diff
@@ -184,6 +177,16 @@ const ManageHostsPage = ({
     return query;
   })();
 
+  const initialPage = (() => {
```

Comment:
Anything with page/initialPage/pageIndex on this file fixes the host page to save the page number when navigating back and forth from host details

Reply:
Moved to other PR that's already merged